media/js/app/taxonomy/taxonomy.js

```diff
@@ -474,16 +494,47 @@
                 return item.display_name == thing;
             });
         },
-        getTheOnomy: function(onomyURL, selectedVocabulary) {
+        getTheOnomy: function(onomyURL, selectedVocabulary, JSON_FLAG) {
```

Comment:
The addition of the skos logic to the getTheOnomy function is causing it to grow again. What do you think of splitting the logic and having a skos importer and  a json importer with some helper functions?